From 779e72d1632f789215510691390a2980b0efef7e Mon Sep 17 00:00:00 2001
From: anatar818 <151121941+anatar818@users.noreply.github.com>
Date: Mon, 3 Aug 2026 20:56:12 +0530
Subject: [PATCH] RDKCOM-5616: RDKBDEV-3470, RDKBACCL-1938,RDKBACCL-1945:
 DCMSetting.conf is empty and in ACS device is not online due to Configurable
 WAN Interface feature (#312)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

…rface

RDKBACCL-1938 : DCMSetting.conf is empty due to Configurable WAN
Interface

Reason for change: After enbaling Configurable WAN , DCM feature is not
working as expected
Test Procedure: Build and flash the image ,validate DAC functionality
with customised wan name
Risks: None
Priority: P2

---------

Co-authored-by: Santosh Nayak <70348540+snayak002c@users.noreply.github.com>
---
 .../TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c |  2 +-
 source-arm/TR-181/board_sbapi/cosa_ip_apis.c          | 11 +++++++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c b/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
index 60a26cb..eb6ef3d 100644
--- a/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
+++ b/custom/comcast/source/TR-181/custom_sbapi/cosa_deviceinfo_apis_custom.c
@@ -214,7 +214,7 @@ CosaDmlDiGetRouterMacAddress
     )
 {
     UNREFERENCED_PARAMETER(hContext);
-#ifdef FEATURE_RDKB_XDSL_PPP_MANAGER
+#if defined(FEATURE_RDKB_XDSL_PPP_MANAGER) || defined(FEATURE_RDKB_CONFIGURABLE_WAN_INTERFACE)
     char wanPhyName[32] = {0};
     char out_value[32] = {0};
 
diff --git a/source-arm/TR-181/board_sbapi/cosa_ip_apis.c b/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
index 55c8033..a797dc1 100644
--- a/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
+++ b/source-arm/TR-181/board_sbapi/cosa_ip_apis.c
@@ -440,6 +440,17 @@ CosaDmlIpInit
 	   g_usg_if_cfg[i].LinkType = ( TRUE == IsThisCurrentPartnerID("sky-") ) ? COSA_DML_LINK_TYPE_VlanLink : COSA_DML_LINK_TYPE_EthLink;   
 	}
 #endif
+ if (i == 0)
+ {
+        
+#if ( defined(FEATURE_RDKB_CONFIGURABLE_WAN_INTERFACE) && !defined( _HUB4_PRODUCT_REQ_ ) )
+        char out_value[64] = {0};
+		if (!syscfg_get(NULL, "wan_physical_ifname", out_value, sizeof(out_value)) && out_value[0] !=0)
+			rc = strcpy_s(g_usg_if_cfg[i].IfName, sizeof(g_usg_if_cfg[i].IfName), out_value);  
+#endif
+	    rc = strcpy_s((char *)g_ipif_names[i],sizeof(g_ipif_names[i]), G_USG_IF_NAME(i));
+ }
+ else
         rc = strcpy_s((char *)g_ipif_names[i],sizeof(g_ipif_names[i]), G_USG_IF_NAME(i));
         ERR_CHK(rc);
         if ((gDmsbIpIfLoopbackInstNum == 0) && (strcmp(G_USG_IF_NAME(i), "lo") == 0))
-- 
2.43.0

